crates/json-abi/src/item.rs

```diff
@@ -502,6 +502,17 @@ impl Function {
         signature(&self.name, &self.inputs, Some(&self.outputs))
     }
 
+    /// Returns this function's full signature including names of params:
+    /// `event $name($($inputs names),*) state_mutability returns ($(outputs names),*)`.
+    ///
+    /// This is a full human-readable string, including all parameter names, any optional modifiers
+    /// (e.g. indexed, public, etc) and white-space to aid in human readability. This is useful for
+    /// storing a string which can still fully reconstruct the original Fragment
+    #[inline]
+    pub fn full_signature(&self) -> String {
```

Comment:
I think we should rename the previous `signature_full` to `signature_with_outputs`

Reply:
I like this idea too. to remove `signature_full` would be a breaking change. Maybe we just add the `signature_with_outputs` as an alias and warn `signature_full` is going to be removed?

(not sure how y'all do SemVer when you're not at 1.0 yet. I guess removing it could come when it becomes 1.0.0!)

Reply:
That's OK, the next release we'll be doing soon is already breaking

Reply:
updated to `signature_with_outputs`